Documentation/planning/multi-sdk-band-support.md

```diff
@@ -0,0 +1,253 @@
+
+This section details changes to the source build infrastructure to support building of subsetted VMRs.
+
+### Scripting changes
+
+Since conceptually CSB and PSB inputs are different, the scripting should reflect this. Furthermore, the since NuGet and non-NuGet archives are treated similarly within the source-build infrastructure, we should take this opportunity to make changes to the names of existing switches:
+
+- `--with-packages` becomes `--with-previous-artifacts`, reflecting that it represents only previously source built artifacts that may be used in the build, but not redistributed with outputs. When poisoning, this set is poisoned.
+- `--with-current-artifacts` - This switch is added. It points to a directory that contains input NuGet and non-NuGet assets that may be redistributed in the current build. Typically this set would come from the 1xx build.
+
+In addition, we should add a script that prepares the input artifacts for the build of a particular VMR branch, given a set of input directories from other builds/other previously source built artifacts. See [below](#what-artifacts-should-be-passed-with---with-artifacts--with-packages) for an explanation of the input artifacts.
+
+### Changes to PackageVersion.props flow
+
+When determining what version of a package or asset that a component should build against, there are 3 distinct sets in source-buidl today, listed in order of precedence from highest to lowest:
+
+- 1. A version just produced by a component built earlier in the build (if available).
+- 2. A version in the previously source-built packages (if available).
+- 3. The version in the checked in `Versions.props` for a repo.
+
+With the addition of a new input set (CSB), we now have an additional set. Changes should be made such that a PackageVersions.props file is generated for this new set, and is imported in the correct place based on its precedence.
+
+- 1. A version just produced by a component built earlier in the build (if available).
+- 2. A version in the current source-built packages (if available).
+- 3. A version in the previously source-built packages (if available).
+- 3. The version in the checked in `Versions.props` for a repo.
+
+### Filtering builds of components
+
+The VMR defines a set of projects (under `repo-projects`) that correspond to each component of the build. These projects also define a build dependency graph of these components via the `RepositoryReference` ItemGroup. *Note: there are a few projects that do not correspond directly to a component, but instead serve as general join points in the build graph.*. Source build invokes a root build (`dotnet.proj`), which evaluates the dependency graph and invokes the MSBuild task on dependent projects. To avoid maintenance complexity, we should avoid altering this dependency graph or removing project files when a component should not be built. Instead, it should simply be filtered out and ignored as if the node in the graph was inactive.
```

Comment:
What are the scenarios that require filtering out the components? Is it just building the subsetted VMRs or something else yet? It would be useful to clarify the motivation for this at the beginning of this section.

Reply:
This is the core way to support multiple-bands band builds, which are represented via subsetted VMRs.

Reply:
Yep, it was more clear from the following text.